File: legacyjava/core/src/main/java/search/TranspositionTable.java
package search;

import org.javafish.move.Move;

import java.util.HashMap;

public class TranspositionTable {
    private final HashMap<Long, TTEntry> table = new HashMap<>();

    public void set(long key, int score, int depth, int flag, Move bestMove){
        table.put(key, new TTEntry(key, score, depth, flag, bestMove));
    }

    public TTEntry probe(long key){
        return table.get(key);
    }

    public void reset(){
        table.clear();
    }
}

File: ts-engine/src/TranspositionTable.ts
import {TTEntry} from './TTEntry';
import {Move} from './Move';

export class TranspositionTable {
  private table: Map<number, TTEntry> = new Map();

  // Method to set an entry in the transposition table
  set(key: number, score: number, depth: number, flag: number, bestMove: Move): void {
    this.table.set(key, new TTEntry(key, score, depth, flag, bestMove));
  }

  // Method to probe (retrieve) an entry from the table
  probe(key: number): TTEntry | undefined {
    return this.table.get(key);
  }

  // Method to reset (clear) the table
  reset(): void {
    this.table.clear();
  }
}
